### PR TITLE
Add a convenience target to locally regenerate workflow files

### PR DESCRIPTION
**Pull request by iwahbe**
This makes it easier to author PRs such as https://github.com/pulumi/pulumi-azuread/pull/490.

### CODE REVIEW

File: provider-ci/internal/pkg/templates/bridged-provider/Makefile [hunk 1]
```diff
@@ -154,4 +154,17 @@ upstream.rebase:
...
+#
+# - Edit .ci-mgmt.yaml
+# - Run make ci-mgmt to apply the change locally.
+#
+ci-mgmt: .ci-mgmt.yaml
+	go run github.com/pulumi/ci-mgmt/provider-ci@master generate \
+		--name pulumi/pulumi-$(PACK) \
+		--out . \
+		--template bridged-provider \
+		--config $<
```
Comment (mjeffryes) [inline provider-ci/internal/pkg/templates/bridged-provider/Makefile:167]: Do we want to emulate the upgrade workflows and wipe the .github/workflows dir before regenerating?

**Review by mjeffryes — APPROVED** [review]

File: provider-ci/internal/pkg/templates/bridged-provider/Makefile (hunk 1 above)
Comment (iwahbe) [reply provider-ci/internal/pkg/templates/bridged-provider/Makefile]: Seems like a good idea.

File: provider-ci/internal/pkg/templates/bridged-provider/Makefile [hunk 2]
```diff
@@ -154,4 +154,18 @@ upstream.rebase:
...
 
-.PHONY: development build build_sdks install_go_sdk install_java_sdk install_python_sdk install_sdks only_build build_dotnet build_go build_java build_nodejs build_python clean cleanup#{{ if .Config.docsCmd }}# docs#{{end}}# help install_dotnet_sdk install_nodejs_sdk install_plugins lint_provider provider test tfgen#{{ if eq .Config.team "ecosystem" }}# upstream upstream.finalize upstream.rebase#{{end}}#
+# To make an immediately observable change to .ci-mgmt.yaml:
+#
+# - Edit .ci-mgmt.yaml
+# - Run make ci-mgmt to apply the change locally.
+#
+ci-mgmt: .ci-mgmt.yaml
+	rm .github/workflows/*.yml # Copied from update-workflows.yml
+	go run github.com/pulumi/ci-mgmt/provider-ci@master generate \
```
Comment (t0yv0) [inline provider-ci/internal/pkg/templates/bridged-provider/Makefile:164]: Hm for a dev target `@master` is probably good.

**Review by t0yv0 — APPROVED** [review]
Yes I need this very much. Thanks.

**Pull request merged**